### PR TITLE
Vim 8.2.4305: tex filetype detection fails; Vim 8.2.4306: no test for fixed perl filetype check

### PR DESCRIPTION
**Pull request by clason**
also sort all tests

### CODE REVIEW

**Review by zeertzjq — APPROVED** [review]
The changed part of the test file is exactly the same as upstream.

**Comment by clason** [comment]
> The changed part of the test file is exactly the same as upstream.

Yes. As far as I could tell, there were no Neovim specific changes to this test file, so taking the sorted upstream list seemed to me less error-prone than manually applying each change, line by line.

I'm happy to be corrected if I missed something, though!

File: src/nvim/testdir/test_filetype.vim [hunk 1]
```diff
@@ -731,101 +731,126 @@ func Test_setfiletype_completion()
...
-  call assert_equal('typescript', &filetype)
+  " Visual Basic
+
+  call writefile(['Attribute VB_NAME = "Testing"'], 'Xfile.bas')
+  split Xfile.bas
+  call assert_equal('vb', &filetype)
   bwipe!
 
-  call delete('Xfile.hook')
+  call delete('Xfile.bas')
```
Comment (zeertzjq) [inline src/nvim/testdir/test_filetype.vim:796]: This comes from patch 8.2.2112

File: src/nvim/testdir/test_filetype.vim [hunk 2]
```diff
@@ -866,141 +891,93 @@ func Test_ex_file()
...
+  call assert_equal('foam', &filetype)
   bwipe!
 
-  " Octave
+  call delete('0', 'rf')
+  call delete('0.orig', 'rf')
+  call delete('Xfile1Dict')
+  call delete('Xfile1Dict.something')
+  call delete('XfileProperties')
+  call delete('XfileProperties.something')
```
Comment (zeertzjq) [inline src/nvim/testdir/test_filetype.vim:944]: This comes from patch 8.2.4200

**Comment by clason** [comment]
Oh, good catch! I missed those two (8.2.4300 was already ported by you). Unfortunately, the other changes in these two patches require other patches to apply.

What should we do? Still merge and just skip `test_filetype.vim` when porting the other two PRs eventually, or skip porting 4306 (which makes porting future filetype updates harder)?

**Comment by zeertzjq** [comment]
> 8.2.4300 was already ported by you

What?

> What should we do? Still merge and just skip `test_filetype.vim` when porting the other two

That sounds better.

**Comment by clason** [comment]
> > 8.2.4300 was already ported by you
> 
> What?

Sorry, typo: I meant 4033 ("running filetype tests leaves directory behind (#16984)")

> > What should we do? Still merge and just skip `test_filetype.vim` when porting the other two
> 
> That sounds better.

+1

**Pull request merged**